src/System.Text.Primitives/System/Text/InvariantParser_byte.cs

```diff
@@ -35,17 +35,22 @@ public static bool TryParse(byte[] utf8Text, int index, out byte value, out int
                         return true; // otherwise return true
                     }
                 }
-                byte candidate = (byte)(value * 10); // left shift the value
-                candidate += (byte)(nextByte - '0'); // parse the current digit to a byte and add it to the temporary value
-                if (candidate >= value) // if it was a digit 0-9, this should be true
+                try
                 {
-                    value = candidate;
+                    byte candidate = checked((byte)(value * 10 + nextByte - '0')); // left shift the value and add the nextByte.
+
+                    if (candidate >= value) // if it was a digit 0-9, this should be true
```

Comment:
Not unless something completely unexpected occurs.

Line 26's if statement makes it so 0 <= `nextByte - '0'` <= 9. If overflow occurs, checked will take care of that. If value is 0 and `nextByte - '0'` is 0, then candidate == value; otherwise, candidate > value.